### PR TITLE
Track empty inputs in reused sandboxes

### PR DESCRIPTION
**Pull request by dzbarsky**
### Description
Track empty files alongside symlinks in the existing compact content map, distinguish the two cases during cleanup, and preserve directories containing only empty inputs.

### Motivation

Null-backed sandbox inputs create real empty files, but in-memory sandbox contents omit those files and their parent directories. A reused sandbox can consequently leak empty inputs between actions or fail to recognize empty inputs that should be retained.

### Build API Changes
No

### Checklist

- [x] I have added tests for the new use cases (if any).
- [ ] I have updated the documentation (if applicable).

### Release Notes

RELNOTES: None

### CODE REVIEW

**Comment by dzbarsky** [comment]
@fmeum @tjgq PTAL!

**Review by fmeum — APPROVED** [review]

**Comment by fmeum** [comment]
@bazel-io fork 9.3.0

**Review by meisterT — APPROVED** [review]

File: src/main/java/com/google/devtools/build/lib/sandbox/SandboxHelpers.java [hunk 1]
```diff
@@ -858,7 +862,7 @@ public static void updateContentMap(Path root, long timestamp, SandboxContents s
         }
         Path absPath = root.getChild(dirent.getName());
         if (dirent.getType().equals(SYMLINK)) {
-          if (stashContents.symlinkMap().containsKey(dirent.getName())
+          if (stashContents.fileMap().get(dirent.getName()) != null
```
Comment (meisterT) [inline src/main/java/com/google/devtools/build/lib/sandbox/SandboxHelpers.java:866]: This came up in internal review:

`updateContentMap` does not appear to handle empty input files stored in `fileMap`.

For empty input files, `fileMap` stores `null` values and the file on disk is a regular file (`FILE`) rather than a `SYMLINK`. When `updateContentMap` checks `dirent.getType().equals(SYMLINK)` and `stashContents.fileMap().get(dirent.getName()) != null`, empty input files evaluate to `false` and fall through to the trailing `else` block, causing `absPath.delete()` to delete the empty file from disk and purge it from `fileMap` (via `filesAndSymlinksToKeep`).

To support empty inputs during `updateContentMap`, consider distinguishing between symlinks and empty files, for example:

```java
} else if (dirent.getType().equals(FILE)) {
  if (stashContents.fileMap().containsKey(dirent.getName())
      && stashContents.fileMap().get(dirent.getName()) == null
      && absPath.stat().getLastChangeTime() <= timestamp) {
    filesAndSymlinksToKeep.add(dirent.getName());
  } else {
    absPath.delete();
  }
}
```

Could you also add a unit test in `SandboxHelpersTest` to verify that `updateContentMap` preserves empty inputs when a directory is updated?
